src/System.Management.Automation/engine/MshObject.cs

```diff
@@ -239,14 +232,8 @@ internal static Collection<CollectionEntry<PSMemberInfo>> GetMemberCollection(PS
                 else
                 {
                     returnValue.Add(new CollectionEntry<PSMemberInfo>(
-                        delegate (PSObject msjObj)
-                        {
-                            return TypeTableGetMembersDelegate<PSMemberInfo>(msjObj, backupTypeTable);
-                        },
-                        delegate (PSObject msjObj, string name)
-                        {
-                            return TypeTableGetMemberDelegate<PSMemberInfo>(msjObj, backupTypeTable, name);
-                        },
+                        msjObj => TypeTableGetMembersDelegate<PSMemberInfo>(msjObj, backupTypeTable),
+                        (msjObj, name) => TypeTableGetMemberDelegate<PSMemberInfo>(msjObj, backupTypeTable, name),
```

Comment:
I think we should write new code using new patterns. If someone wants to clean up older code, fine. But I would really dislike having to use some old construct just because some other old code hasn't been updated.